
Training T-S model on chest
chest Training:   0%|                                                             | 0/450 [00:00<?, ?it/s]
chest epochs:   0%|                                                            | 0/100 [00:00<?, ?epoch/s]
Traceback (most recent call last):
  File "/home/zeddo123/anom/RD4AD/easy_train.py", line 122, in <module>
    test(args, c)
  File "/home/zeddo123/anom/RD4AD/easy_train.py", line 74, in test
    for imgs, _, labels, class_defect, gndtruth in tqdm(train_loader, desc=f'{_class_} Training'):
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/tqdm/std.py", line 1181, in __iter__
    for obj in iterable:
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 673, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "/home/zeddo123/anom/RD4AD/easy_dataset.py", line 133, in __getitem__
    return self._prepare_item(index)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/easy_dataset.py", line 118, in _prepare_item
    img = self.transform(img)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torchvision/transforms/transforms.py", line 95, in __call__
    img = t(img)
          ^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torchvision/transforms/transforms.py", line 277, in forward
    return F.normalize(tensor, self.mean, self.std, self.inplace)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torchvision/transforms/functional.py", line 350, in normalize
    return F_t.normalize(tensor, mean=mean, std=std, inplace=inplace)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zeddo123/anom/RD4AD/venv/lib/python3.12/site-packages/torchvision/transforms/_functional_tensor.py", line 926, in normalize
    return tensor.sub_(mean).div_(std)
           ^^^^^^^^^^^^^^^^^
RuntimeError: output with shape [1, 256, 256] doesn't match the broadcast shape [3, 256, 256]